### ISSUE TITLE
Operators not forwarded through `public use` / `import`?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#19253]
It seems that operators are not forwarded through `public use` or `import` statements.  For example, in the following program if the `use N` is replaced with `use M`, things work.  But the operators are not seeming to be seen through the public use (import seems to work no better).

```chapel
module M {
  extern type c_yaddayadda = opaque;

  operator c_yaddayadda.:(x: borrowed, type t: c_yaddayadda) {
    return 42;
  }

  operator c_yaddayadda.+(x: c_yaddayadda, y: c_yaddayadda) {
    return 45;
  }
}

module N {
  public use M only c_yaddayadda;
}

module Main {
  class C {
  }
  proc main() {
    var myC = new C();
    use N;
    var x: c_yaddayadda;
    writeln(x + x);
    writeln(myC.borrow():c_yaddayadda);
  }
}
```

**Comment by bradcray** [comment]
@lydia-duncan : Can you check me on this?

**Comment by lydia-duncan** [comment]
Yeah, there's definitely a few things wrong about that, but I think it's more general than operators.

I agree that the name of the type should bring in those operator methods (could've sworn I had a test for that but it might not use aliases).  It does look like it fails to bring in regular methods as well so I suspect the problem might have something to do with type aliases or with extern types.

It also looks as though naming an operator method's name behaves the same as naming a regular method's name in the scope.  So at least it's consistent, but I had thought naming methods in limitation clauses didn't work at all.

I think there's a question about whether an operator defined as a method on a type alias should count as a primary method for the type alias or a tertiary method for the original type if it wasn't defined in the same scope as the type alias.  I think today it really counts as a tertiary method and that makes sense since we don't view the alias as a distinct type capable of having methods that the original type does not.  But that might be confusing the resolver - if we've replaced the alias with the type it references already, the compiler might not think to go looking for the original type name.

**Comment by bradcray** [comment]
> I think there's a question about whether an operator defined as a method on a type alias should count as a primary method for the type alias or a tertiary method for the original type if it wasn't defined in the same scope as the type alias.

Hmm, that's an interesting point that hadn't occurred to me.  I'd expect that if the type is an extern type alias, then we should consider anything declared in the scope that it's declared in to be secondary rather than tertiary since it's the closest thing we have to the type's definition point within Chapel.

**Comment by bradcray** [comment]
@lydia-duncan : Do you have a rough estimate on what the level of effort would be to resolve this?

**Comment by lydia-duncan** [comment]
Anywhere from a week's worth to potentially 3 weeks.  The main thing is that I strongly suspect important information about the type alias has been lost by the time we're resolving calls involving instances of it, which means we don't traverse the use/import when we should.  I suspect the majority of the work will be figuring out how to maintain this information and use it without overly complicating use/import traversal in function resolution.
